src/lightning/pytorch/callbacks/progress/rich_progress.py

```diff
@@ -173,8 +180,10 @@ def render(self, task: "Task") -> Text:
             return Text(text, justify="left", style=self._style)
 
         def _generate_metrics_texts(self) -> Generator[str, None, None]:
-            for k, v in self._metrics.items():
-                yield f"{k}: {round(v, 3) if isinstance(v, float) else v}"
+            for name, value in self._metrics.items():
+                if not isinstance(value, str):
+                    value = round(value, 3) if self._metrics_format is None else f"{value:{self._metrics_format}}"
```

Comment:
`metrics_format` could have `.3f` as default which should be equivalent to `round(, 3)`:

Reply:
I definitely agree. It would not entirely be backward compatible for metrics with lower precision. That's why I didn't propose it originally.

e.g.
100.1 becomes
* round(, 3) => 100.1
* .3f => 100.100

In my opinion, the .3f option is actually better than the current behavior because it results in consistent metrics lengths, making the progress bar values jump around less, and making the metrics more readable. (For example, if 10.111 changes to 10.1, this introduces a shift in all other metric values)

Reply:
Good point. I don't think we need to be this strict about backward compatibility in terms of how the data is presented on the progress bar. Especially if the change is it's for a better UI

Reply:
Okay, I agree. I created a PR for this at https://github.com/Lightning-AI/lightning/pull/18483